ENH: testing: hide assertion internals in pytest traceback

Trivial one, but: when using `xpx.testing`, pytest shows the failed asserts within the xpx library code, not at the call site.

```py
# test_xpx.py
import numpy as np
import array_api_extra as xpx

def test_less():
    xpx.testing.assert_less(np.arange(3), np.arange(3))
```

```console
% pytest -xvv test_xpx.py
==================================== test session starts ====================================
...

test_xpx.py::test_less FAILED                                                         [100%]

========================================= FAILURES ==========================================
_________________________________________ test_less _________________________________________

    def test_less():
>       xpx.testing.assert_less(np.arange(3), np.arange(3))

test_xpx.py:5:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

x = array([0, 1, 2]), y = array([0, 1, 2])

    def assert_less(
        x: Array,
        y: Array,
        *,
        err_msg: str = "",
        verbose: bool = True,
        check_dtype: bool = True,
        check_shape: bool = True,
        check_scalar: bool = False,
        xp: ModuleType | None = None,
    ) -> None:
        ...
        (the entire function's source code here)
        ...
>       np.testing.assert_array_less(x_np, y_np, err_msg=err_msg, verbose=verbose)
E       AssertionError:
E       Arrays are not strictly ordered `x < y`
E
E       Mismatched elements: 3 / 3 (100%)
E       Max absolute difference among violations: 0
E       Max relative difference among violations: 0.
E        x: array([0, 1, 2])
E        y: array([0, 1, 2])

.../site-packages/array_api_extra/testing.py:928: AssertionError
================================== short test summary info ==================================
FAILED test_xpx.py::test_less - AssertionError:
Arrays are not strictly ordered `x < y`

Mismatched elements: 3 / 3 (100%)
Max absolute difference among violations: 0
Max relative difference among violations: 0.
 x: array([0, 1, 2])
 y: array([0, 1, 2])
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
===================================== 1 failed in 0.24s =====================================
```

--- a/src/array_api_extra/testing.py
+++ b/src/array_api_extra/testing.py
@@ -770,6 +770,7 @@ def assert_close(
 
     Array arguments to `atol` and `rtol` must be valid input to :class:`float`.
     """
+    __tracebackhide__ = True
     actual, desired, xp, np = _check_ns_shape_dtype(
         actual, desired, check_dtype, check_shape, check_scalar, xp
     )
@@ -850,6 +851,7 @@ def assert_equal(
     assert_close : Similar function for inexact equality checks.
     numpy.testing.assert_array_equal : Similar function for NumPy arrays.
     """
+    __tracebackhide__ = True
     actual, desired, xp, np = _check_ns_shape_dtype(
         actual, desired, check_dtype, check_shape, check_scalar, xp
     )
@@ -908,6 +910,7 @@ def assert_less(
     assert_close : Similar function for inexact equality checks.
     numpy.testing.assert_array_less : Similar function for NumPy arrays.
     """
+    __tracebackhide__ = True
     x, y, xp, np = _check_ns_shape_dtype(
         x, y, check_dtype, check_shape, check_scalar, xp
     )

